### ISSUE TITLE
Improve the API 

### ISSUE THREAD

**Issue by maltezfaria** [maltezfaria/DataFlowTasks.jl#6]

**Comment by ffevotte** [comment]
FYI: I'm working on something that would look like this:

```
julia> using DataFlowTasks: @dspawn_new

julia> let
           x = rand(1000)
       
           t1 = @dspawn_new begin
               @W x
               sleep(1)
               x .= 0.001
           end
       
           t2 = @dspawn_new sum(@R x)
       
           fetch(t2)
       end
1.0000000000000007
```

where macro calls expand to things like:
```
julia> @macroexpand @dspawn_new begin
           @RW x
           @R  y z
           x .= y .+ z
       end priority=2
quote
    var"#279#t" = (DataFlowTasks.DataFlowTask)((()->begin
                    begin
                        x
                        y
                        x .= y .+ z
                    end
                end), (x, y, z), (DataFlowTasks.READWRITE, DataFlowTasks.READ, DataFlowTasks.READ), 2)                               
    (DataFlowTasks.spawn)(var"#279#t")
end
```

**Comment by ffevotte** [comment]
A side effect of this is that there is no need anymore to import things like `DataFlowTasks.R`, `W` and `RW`. I think is rather nice, but that can cause collisions if there exist `@R`, `@W` or `@RW` macros coming from some user-defined or third-party code.

**Comment by maltezfaria** [comment]
I think the risk of collisions is acceptable -- not too many people define macros :-) 

And I like the proposed `API` !

One thing that would be nice as well is to implement a global way to deactivate the parallelism and execute the code sequentially.

**Comment by maltezfaria** [comment]
closed by #9
